### PR TITLE
Service Methods Reset

### PR DESCRIPTION
**Pull request by Zarathustra4**

### CODE REVIEW

File: src/main/java/com/passpnu/passwordmanager/mapper/ServiceMapper.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package com.passpnu.passwordmanager.mapper;
+
+import com.passpnu.passwordmanager.dto.ServiceDto;
+import com.passpnu.passwordmanager.entity.ServiceEntity;
+import org.mapstruct.Mapper;
+import org.springframework.stereotype.Component;
+
+@Mapper
```
Comment (justme1135) [inline src/main/java/com/passpnu/passwordmanager/mapper/ServiceMapper.java]: First of all, mapstruct, should handle these conversion automatically, you don't need to write mappings manually.
Second of all, Your mapper should be an interface with two abstract method like you already have got

**Review by justme1135 — CHANGES_REQUESTED** [review]

File: src/main/java/com/passpnu/passwordmanager/mapper/ServiceMapper.java (hunk 1 above)
Comment (Zarathustra4) [reply src/main/java/com/passpnu/passwordmanager/mapper/ServiceMapper.java]: Fixed

File: src/main/java/com/passpnu/passwordmanager/service/ServiceEntityServiceImpl.java [hunk 2]
```diff
@@ -14,67 +15,48 @@
...
     @Override
     public ServiceDto putService(ServiceDto service) throws NameNotFoundException {
-        Optional<ServiceEntity> serviceOptional = Optional.ofNullable(serviceRepository.findByDomain(
+        ServiceEntity serviceEntity = serviceRepository.findByDomain(
                 service.getDomain()
-        ));
+        );
 
-        if(serviceOptional.isEmpty()){
+        if(serviceEntity == null){
```
Comment (justme1135) [inline src/main/java/com/passpnu/passwordmanager/service/ServiceEntityServiceImpl.java]: Rewrite using `Objects.requireNonNull`

**Review by justme1135 — CHANGES_REQUESTED** [review]

File: src/main/java/com/passpnu/passwordmanager/service/ServiceEntityServiceImpl.java [hunk 3]
```diff
@@ -85,11 +67,11 @@ public ServiceDto getById(String id) throws NameNotFoundException{
             throw new NameNotFoundException("The service does not exist");
         }
 
-        return entityToDto(serviceEntityOptional.get());
+        return serviceMapper.entityToDto(serviceEntityOptional.get());
     }
 
     @Override
     public boolean existsByDomain(ServiceDto service){
         return serviceRepository.existsByDomain( service.getDomain() );
```
Comment (getmane) [inline src/main/java/com/passpnu/passwordmanager/service/ServiceEntityServiceImpl.java:68]: Remove spaces

Comment (Zarathustra4) [reply src/main/java/com/passpnu/passwordmanager/service/ServiceEntityServiceImpl.java:68]: fixed

File: src/main/java/com/passpnu/passwordmanager/service/ServiceEntityServiceImpl.java (hunk 2 above)
Comment (Zarathustra4) [reply src/main/java/com/passpnu/passwordmanager/service/ServiceEntityServiceImpl.java]: fixed

**Review by justme1135 — APPROVED** [review]

**Review by getmane — APPROVED** [review]

**Pull request merged**